### PR TITLE
Add searchbar to the top of stack template section, to search stack scripts

### PR DESCRIPTION
**Pull request by hakankaradis**
Aim of this pr is giving chance to user to search and preview the online stack scripts from Stack-Editor/stack-templates view.
## Description

Users will be able to search and read stack scripts
## Motivation and Context

Users will be able to see stack script examples for any provider, without leaving the site.
## How Has This Been Tested?

Go to stack-editor page and search for stack scripts.
## Screenshots (if appropriate):

http://recordit.co/0jZpzFFqKN
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

/cc: @ftasdelen @sinan

### CODE REVIEW

**Comment by gokmen** [comment]
@hakankaradis can we tweak this UI a bit; make sure `1` covers `2` 

image 

but not move out the boundries. and show the result in place instead of opening a new modal. cc/ @ftasdelen @sinan

File: client/app/lib/flux/environment/actions.coffee [hunk 1]
```diff
@@ -828,6 +829,21 @@ loadExpandedMachineLabel = (label) ->
...
+
+  new Promise (resolve, reject) ->
+    options =
+      type: 'GET'
+      contentType: 'application/json'
+      url: url
+      success: (data) -> resolve JSON.parse data
+      error: (jqXHR, textStatus, errorThrown) -> reject()
+
+    $.ajax options
```
Comment (usirin) [inline client/app/lib/flux/environment/actions.coffee]: We can use `makeHttpClient` util for this one.
```coffee
makeHttpClient = require 'app/util/makeHttpClient'

client = makeHttpClient { baseURL: '/-/stack' }

client.get("/getContent/#{query}").then({ data }) ->
  ...
```

**Comment by hakankaradis** [comment]
@gokmen http://recordit.co/yRwuSsYmlq here is the final design
/cc: @ftasdelen

**Comment by gokmen** [comment]
@hakankaradis looks better, can we remove this modal usage? or at least can we center the modal on stack editor instead of whole page and lastly can u add spacing here at the bottom of result list;

image

cc/ @ftasdelen @sinan

**Comment by ftasdelen** [comment]
@hakankaradis We can center align the modal. Good feedback @gokmen

**Comment by hakankaradis** [comment]
@gokmen We decided to keep using modal. So I am going to center the modal to stack-editor and add decrease modal max-height to give space between modal and buttons which are preview and logs. @ftasdelen

**Comment by hakankaradis** [comment]
@gokmen http://recordit.co/cqoORZzZWD

**Review by gokmen — APPROVED** [review]

**Pull request merged**